src/map/clif.c

```diff
@@ -10657,6 +10657,12 @@ static void clif_parse_LoadEndAck(int fd, struct map_session_data *sd)
 		clif->updatestatus(sd, SP_CARTINFO);
 	}
 
+	// Unequip items which can't be equipped by the character.
+	for (int i = 0; i < EQI_MAX; i++) {
```

Comment:
Maybe it could be worth running this only on login, rather than on every map change (i.e. `if (sd->state.connect_new != 0)`)

Reply:
Woops. My bad. It was intended to add this to the `if (sd->state.connect_new != 0)` branch.
*fixed*

Reply:
Uh no it's intended to happen in every map change, otherwise zone item restrictions won't be checked.